OpenEdXMobile/src/main/java/org/edx/mobile/view/adapters/CourseOutlineAdapter.java

```diff
@@ -360,6 +360,9 @@ private void getRowViewForLeaf(ViewHolder viewHolder, final SectionRow row) {
             }
             checkAccessStatus(isDenialFeatureBasedEnrolments, viewHolder, unit);
         }
+        if (unit.getType() == BlockType.OPENASSESSMENT) {
+            viewHolder.rowType.setIcon(FontAwesomeIcons.fa_edit);
+        }
```

Comment:
Why have you separated out this `if-block`? Shouldn't it come:
- after `else if (config.isDiscussionsEnabled() && row.component instanceof DiscussionBlockModel) {`?
- and before `else if (!unit.isMultiDevice()) {`?

Reply:
`BlockType.OPENASSESSMENT` can fall into more than one condition so handled it separately.